tests/queries/test_query.py

```diff
@@ -91,6 +91,20 @@ def test_negated_nullable(self):
         self.assertIsInstance(lookup, IsNull)
         self.assertEqual(lookup.lhs.target, Item._meta.get_field('modified'))
 
+    def test_negated_nullable_with_excluding_on_annotation(self):
+        # nullable field
+        query = str(Item.objects.annotate(year=ExtractYear('modified')).exclude(year='2021').values('id').query)
+        self.assertTrue('IS NOT NULL' in query)
```

Comment:
I think evaluating the result of a query (such as with `assertQuerysetEqual`) is much better than checking whether there's a random IS NOT NULL somewhere in the SQL. This would involve creating some objects.

Reply:
@jacobtylerwalls okay. Thanks for your review.